### PR TITLE
*: only disable retry for write conflict

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
If `set @@tidb_disable_txn_auto_retry = 1`
it will disable retry for any reasons, sometimes the user wants to disable it only for the write conflict.

### What is changed and how it works?
Make `tidb_disable_txn_auto_retry` disable retry only for write conflict.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

**Comment by jackysp** [comment]
Yes, I've modified the description. @zimulala

**Comment by tiancaiamao** [comment]
LGTM

File: session/session.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func (s *session) doCommitWithRetry(ctx context.Context) error {
 			// For autocommit single statement transactions, the history count is always 1.
 			// For explicit transactions, the statement count is more than 1.
 			history := GetHistory(s)
-			if history.Count() > 1 {
+			if history.Count() > 1 && strings.Contains(err.Error(), util.WriteConflictMarker) {
```
Comment (lysu) [inline session/session.go]: can this be replaced with `strings.HasPrefix`?

Comment (lysu) [reply session/session.go]: and I have another irrelevant question~

If we have  a tx with two statements, select then insert... maybe we should not  retry insert stmt too? event if `history.count() == 1`

Comment (jackysp) [reply session/session.go]: Yes, this condition is not correct.

Comment (jackysp) [reply session/session.go]: I've tested it. It is correct...

**Comment by tiancaiamao** [comment]
PTAL @lysu

**Pull request merged**